### PR TITLE
Static type checking

### PR DESCRIPTION
**Pull request by SamHellawell**
Adds static type checking using JSDoc, helps to address https://github.com/docknetwork/sdk/issues/89

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,8 +68,9 @@ yarn test-with-node
 
 Run a single test module with `yarn test <module name or prefix>`.
 
-## Lint
-`yarn lint --fix`
+## Linting and type checking
+We use JSDoc and TypeScript for static type checking during the CI process. PRs must pass linting and type checking to be accepted.
+You can run via: `yarn lint --fix` and `yarn type-check`
```
Comment (lovesh) [inline README.md:73]: Would `type-check` fix the lint errors as well?

File: src/utils/vc/document-loader.js [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 import testContext from './contexts';
+import DIDResolver from '../../did-resolver'; // eslint-disable-line
 
 /**
  * Takes a resolver and returns a function that returns a document or throws an error when the document
  * cannot be found.
- * @param {Resolver} resolver - The resolver is optional but should be passed when DIDs need to be resolved.
- * @returns {function(*=): {documentUrl: *, document: null}}
+ * @param {DIDResolver} [resolver] - The resolver is optional but should be passed when DIDs need to be resolved.
+ * @returns {loadDocument} - the returned function
```
Comment (lovesh) [inline src/utils/vc/document-loader.js:9]: Shouldn't `loadDocument` be function?

File: src/utils/vc/helpers.js [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+ */
+
 /**
  * Helper to get the key doc in a format needed for vc.js.
  * @param {string} did - DID in fully qualified form
  * @param {object} keypair - Keypair is generated by either using polkadot-js's keyring for Sr25519 and
  * Ed25519 or keypair generated with `generateEcdsaSecp256k1Keypair` for curve secp256k1.
  * @param {string} type - the type of key, Sr25519VerificationKey2020 or Ed25519VerificationKey2018 or EcdsaSecp256k1VerificationKey2019
- * @returns {{publicKeyBase58: *, controller: *, id: string, type: *, privateKeyBase58: (string|KeyObject|T2|Buffer|CryptoKey)}}
+ * @returns {KeyDoc}
```
Comment (lovesh) [inline src/utils/vc/helpers.js:18]: `KeyDoc` -> `function`

File: README.md (hunk 1 above)
Comment (SamHellawell) [reply README.md:73]: nope, they have to be ran separately. we could concat into one command but then hard to see exactly where things go wrong in ci

File: src/utils/vc/document-loader.js (hunk 2 above)
Comment (SamHellawell) [reply src/utils/vc/document-loader.js:9]: it is a function, this defines that it returns the loadDocument function (that is referenced below at its definition)

File: src/utils/vc/helpers.js (hunk 3 above)
Comment (SamHellawell) [reply src/utils/vc/helpers.js:18]: there is a typedef for KeyDoc https://github.com/docknetwork/sdk/pull/91/files/883a500eb396263547fd40b5e865bfca2e264a00#diff-cbda9485de00c1a8f3f71fac0d2ed188

**Pull request merged**